libr/bin/p/bin_mach0.c

```diff
@@ -451,23 +451,22 @@ static RList *relocs(RBinFile *bf) {
 }
 
 static RList *libs(RBinFile *bf) {
-	int i;
-	char *ptr = NULL;
-	struct lib_t *libs;
-	RList *ret = NULL;
 	RBinObject *obj = bf ? bf->o : NULL;
+	if (!obj) {
+		return NULL;
+	}
 
-	if (!obj || !obj->bin_obj || !(ret = r_list_newf (free))) {
+	const RPVector *libs = MACH0_(load_libs) (obj->bin_obj);
+	if (!libs) {
 		return NULL;
 	}
-	if ((libs = MACH0_(get_libs) (obj->bin_obj))) {
-		for (i = 0; !libs[i].last; i++) {
-			ptr = strdup (libs[i].name);
-			r_list_append (ret, ptr);
-		}
-		free (libs);
+
+	RList *result = r_list_new ();
+	void **it;
```

Comment:
will be good to make r_pvector api foreach() work like the r_vector_foreach one. for consistency, also maybe we can have an r_pvector_tolist() function that does exactly this, because we will need to duplicate this code in a bunch of places if we want to use vector/pvector internally, but keep the public abi with rlist

Reply:
Yeah that's a good point, I didn't even realize the foreach worked so differently for the 2 variations. 😅

Sure I can make a "tolist" function, could be useful in other places too!